src/Amqp/Akka.Streams.Amqp.V1/AmqpSourceStage.cs

```diff
@@ -51,10 +69,45 @@ public override void PreStart()
             {
                 base.PreStart();
 
-                var consumerCallback = GetAsyncCallback<Message>(HandleDelivery);
-                _receiver.Start(_amqpSourceSettings.Credit, (_, m) => consumerCallback.Invoke(m));
+                Connect().Wait();
```

Comment:
Connect call is blocking here but there isn't much you can do about that - have to wait for the execution to finish somewhere in order to start the message pump.

Reply:
Actually, come to think of it - might want to just have the async state machine invoke a callback on the stage and signal failure / success. Given that this retry + backoff mechanism can run for several minutes, you don't want to tie the process up for that long. How hard would it be to do that?

Reply:
I'll give it a look see, it shouldn't be that hard

Reply:
On second thought, can we separate this into a different PR?
It "works" as it is, I'll work on the thread blocking issue on another PR.

Reply:
That's fine with me